### ISSUE TITLE
Page Not Loading - #21 Revisited

### ISSUE THREAD

**Issue by CaptainPackers** [kromitgmbh/titra#135]
I have the same issue that was reported in Issue #21, however the root_url in my docker-compose.yml file does point to the server ip address. I am running Titra in a virtual machine, and it works fine as long as I am on the host or another virtual machine on the host. If I try to access Titra across my LAN from a different device I get the perpetual page loading page. My host is port forwarding to the guest VM running Titra. I am using the same configuration for Wekan and it works fine, so I think my environment is OK. I'm wondering if there is something in the nginx container that is causing this problem, but I'm not familiar enough with docker to dig into it. Any suggestions you can offer that might help would be greatly appreciated. I love Titra and the way it integrates with Wekan. These two tools really have improved my workflow and help greatly with reporting my time to my supervisor. I would really like to use Titra from across my LAN from time to time and not have to resort to a hack like vnc to be able to use titra remotely.

**Comment by faburem** [comment]
Hi,
yes your nginx configuration is likely the root cause, it has to be configured as reverse proxy handling multiple requests and directing it to different backend endpoints - I am also not an expert but I think this will not work with an IP address only, you will need a domain with different subdomains to make this work e.g. wekan.yourhost.xy and titra.yourhost.xy (at least this is also how we have set it up internally). You could consider moving to [traefik](https://www.digitalocean.com/community/tutorials/how-to-use-traefik-v2-as-a-reverse-proxy-for-docker-containers-on-ubuntu-20-04) as reverse proxy if you are only working with docker containers. The learning curve is a bit steep in the beginning but it is optimised for exactly this usage scenario plus it gives you out of the box letsencrypt certificates for https.

I hope this helps,
Fabian

**Comment by CaptainPackers** [comment]
Thanks for the reply. I was going to start looking at traefik (eventually since you said the learning curve was steep). Then I saw [this post on TechRepublic](https://www.techrepublic.com/article/using-virtualbox-vms-on-your-networks-subnet) that mentioned the use of a bridged network adapter as being necessary to access your VMs on you LAN. Previously I had been using a NAT adapter and port forwarding from my host to my guest VM. This worked fine with Wekan, but for some reason it never did work with Titra. 

The bridged adapter puts the guest VM on the same subnet as all the other devices on my LAN, so after configuring my guest VM with a static IP address and changing the root_url in the titra docker-compose.yml file, I was able to connect directly to titra directly from any device on my LAN.

I do appreciate your reply. It gave me a glimmer of hope that there was something else I could try before giving up on titra completely (my last resort was going to be [NextCloud Time Tracker](https://apps.nextcloud.com/apps/timetracker).
